src/mongo.js

```diff
@@ -0,0 +1,266 @@
+      if (result.result.n === 1 && result.result.ok === 1) {
+        return;
+      }
+      throw new ApiError(httpStatus.INTERNAL_SERVER_ERROR);
+    } catch (error) {
+      logError(error);
+      throw new ApiError(httpStatus.INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  function createBulk({correlationId, cataloger, operation, contentType, recordLoadParams, stream}) {
+    const time = moment().toDate();
+    const newQueueItem = {
+      correlationId,
+      cataloger,
+      operation,
+      contentType,
+      recordLoadParams,
+      queueItemState: QUEUE_ITEM_STATE.UPLOADING,
+      creationTime: time,
+      modificationTime: time,
+      handledIds: []
+    };
+    try {
+      db.collection('queue-items').insertOne(newQueueItem);
+      logger.log('info', 'New queue item has been made!');
+    } catch (error) {
+      logError(error);
+      throw new ApiError(httpStatus.INTERNAL_SERVER_ERROR);
+    }
+
+    return new Promise((resolve, reject) => {
+      const outputStream = gridFSBucket.openUploadStream(correlationId);
+
+      stream
+        .on('error', reject)
+        .on('data', chunk => outputStream.write(chunk))
+        .on('end', () => outputStream.end(undefined, undefined, () => {
+          resolve(correlationId);
+        }));
+    });
+  }
+
+  async function checkAndSetState({correlationId, state}) {
+    const timeOut = await checkTimeOut(correlationId);
+    if (timeOut) {
+      return setState({correlationId, state});
+    }
+    return false;
+  }
+
+  async function query(params) {
+    const result = await db.collection('queue-items').find(params, {projection: {_id: 0}})
+      .toArray();
+    logger.log('info', `Query result: ${result.length > 0 ? 'Found!' : 'Not found!'}`);
+    return result;
+  }
+
+  async function queryById(correlationId, checkModTime) {
+    const result = await db.collection('queue-items').findOne({correlationId});
+    if (checkModTime) {
+      const timeOut = await checkTimeOut(correlationId);
+      if (timeOut) {
+        return queryById(correlationId);
+      }
+      return result;
+    }
+
+    return result;
+  }
+
+  async function checkTimeOut(correlationId) {
+    const result = await db.collection('queue-items').findOne({correlationId});
+    const timeoutTime = moment(result.modificationTime).add(1, 'm');
+    logger.log('silly', `timeOut @ ${timeoutTime}`);
+    if (timeoutTime.isBefore()) {
+      await setState({correlationId, state: PRIO_QUEUE_ITEM_STATE.ABORT});
+      return false;
+    }
+
+    return true;
+  }
+
+  async function remove(correlationId) {
+    logger.log('info', `Removing id: ${correlationId}`);
+    try {
+      await getFileMetadata({gridFSBucket, filename: correlationId});
+      throw new ApiError(httpStatus.BAD_REQUEST, 'Remove content first');
+    } catch (err) {
+      if (err.status === httpStatus.NOT_FOUND) { // eslint-disable-line functional/no-conditional-statement
```

Comment:
Disabling eslint here should not be necessary since you do have a return statement.